### PR TITLE
Timer2

### PR DESCRIPTION
**Pull request by winescout**
Discussion
===============
Our Time tracker did not have the notion of a running timer.  We've only been able to input hours after they are completed.  This made it difficult to create a desktop/phone tool to keep track of hours like Toggl does it.  A Toggle workflow is to start a timer with the required information, then at some future point, stop the timer to create an entry.

This feature enables mobile clients that can check this api.  If there is no running timer, give the user a chance to start one.  If there is a timer, give user a chance to stop it.

In this PR
===========
* Adds a Timer model
* Adds a Timer tab to the Project Index where you can create a new timer
* Adds ability to edit/stop a timer and create an hours entry

## Timer on Project Page
Screen Shot 2019-08-11 at 2 36 24 PM

## Managing a running timer
Screen Shot 2019-08-11 at 2 35 58 PM

## A Video
[ScreenFlow.mp4.zip](https://github.com/notch8/Hours/files/3490464/ScreenFlow.mp4.zip)

### CODE REVIEW

**Comment by alishaevn** [comment]
Once we've transitioned to the "your running timer" tab, the fields at the bottom are still editable correct? (It looks like they are, just verifying)

**Comment by alishaevn** [comment]
Also, the time that elapsed from you beginning and ending the "testing timer 4" entry was about 25 seconds (I timed it on my phone). However, the entry shows it as 0.9 hours, which is 54 minutes.

**Comment by winescout** [comment]
That's a good catch Alisha.  I actually updated the start time, so the duration was more than 10 minutes.  Less than that, and it won't create a log for you.

File: docker-compose.yml [hunk 1]
```diff
@@ -41,6 +42,7 @@ services:
     tty: true
     ports:
       - "${DEV_PORT}"
+      - "8082:3000"
```
Comment (orangewolf) [inline docker-compose.yml]: I even left you the DEV_PORT here ;-)
put `DEV_PORT=8082:3000` in .env.development

File: docker-compose.yml [hunk 2]
```diff
@@ -24,6 +24,7 @@ services:
       dockerfile: Dockerfile.base
 
   web:
+    env_file: .env
```
Comment (orangewolf) [inline docker-compose.yml]: this is already 3 lines down

File: spec/controllers/timers_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+describe TimersController, type: :controller do
+
```
Comment (orangewolf) [inline spec/controllers/timers_controller_spec.rb]: remove this empty file please

File: spec/features/owner_deletes_account_spec.rb [hunk 4]
```diff
@@ -7,19 +7,6 @@
   end
 
   context "as the owner of an account" do
-    scenario "deleting the account" do
```
Comment (orangewolf) [inline spec/features/owner_deletes_account_spec.rb]: are these just failing or did we remove this feature?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -28,7 +28,7 @@ module Features
     c.syntax = :expect
   end
 
-  config.fail_fast = true
+  #config.fail_fast = true
```
Comment (orangewolf) [inline spec/spec_helper.rb]: did you mean to commit this?

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -1,6 +1,5 @@
 /* global Pikaday, Tagger, TagExpander */
 
-//= require jquery
```
Comment (orangewolf) [inline app/assets/javascripts/application.js]: I'm curious, why did you move jquery out of the pipeline and in to the layout?

Comment (winescout) [reply app/assets/javascripts/application.js]: you can't have $().ready if Jquery is loaded at the bottom of the page, after the code you need to initialize.

The date/time picker needed it

File: spec/spec_helper.rb (hunk 5 above)
Comment (winescout) [reply spec/spec_helper.rb]: I did.  That stops the run on the first fail.  I generally want to see them all, and then run individual files when I have need for speed

File: spec/features/owner_deletes_account_spec.rb (hunk 4 above)
Comment (winescout) [reply spec/features/owner_deletes_account_spec.rb]: That one was an oddity, failing due to a missing relation of timers, which is odd because there are new model specs for Timer, so....?

File: app/assets/javascripts/application.js (hunk 6 above)
Comment (winescout) [reply app/assets/javascripts/application.js]: I needed document.ready to initialize the time/date picker, and that's not available with Jquery loaded at the end of the page
